src/core_plugins/metric_vis/public/metric_vis_controller.js

```diff
@@ -24,12 +30,72 @@ module.controller('KbnMetricVisController', function ($scope, $element, Private)
 
         metrics.push({
           label: column.title,
-          value: value
+          value: value,
+          aggConfig: table.aggConfig(column)
         });
       });
     });
   };
 
+  $scope.clickHandler = function (metric) {
+    let aggConfigResult = null;
+    if (metric.aggConfig.getField() && metric.aggConfig.getField().filterable) {
+      const fieldAggConfig = {
+        schema: {
+          group: 'buckets'
+        },
+        fieldFormatter: metric.aggConfig.fieldFormatter,
+        createFilter: (key) => {
+          return {
+            exists: {
+              field: metric.aggConfig.getField().name
+            },
+            meta: {
+              index: $scope.vis.indexPattern.id
+            }
+          };
+        }
+      };
+      aggConfigResult = new AggConfigResult(fieldAggConfig, aggConfigResult, metric.aggConfig.getField(), 'exists');
+    }
+    if ($scope.vis.getFilters().length > 0) {
+      $scope.vis.getFilters().forEach((filter) => {
+        const fieldAggConfig = {
+          schema: {
+            group: 'buckets'
+          },
+          fieldFormatter: () => {},
+          createFilter: (key) => {
+            return filter;
+          }
+        };
+        aggConfigResult = new AggConfigResult(fieldAggConfig, aggConfigResult, filter, 'filter');
+      });
+    }
+
+    if (_.get($scope.vis.getQuery(), 'query_string.query', '*') !== '*') {
```

Comment:
what should this achieve ? what is the reason for putting the query inside filter ? its not the pattern kibana would use elsewhere, so this should probably behave the same. Please try to explain the use case for this. In my opinion we could do without.